packages/elements/src/internals/machines/sign-in/verification.machine.ts

```diff
@@ -23,31 +23,44 @@ import { sendToLoading } from '~/internals/machines/shared';
 import { determineStartingSignInFactor, determineStartingSignInSecondFactor } from '~/internals/machines/sign-in/utils';
 import { assertActorEventError, assertIsDefined } from '~/internals/machines/utils/assert';
 
-import type { TSignInRouterMachine } from './router.machine';
+import type { SignInRouterMachineActorRef } from './router.types';
 import type { SignInVerificationSchema } from './verification.types';
 import { SignInVerificationDelays } from './verification.types';
 
 export type TSignInFirstFactorMachine = typeof SignInFirstFactorMachine;
 export type TSignInSecondFactorMachine = typeof SignInSecondFactorMachine;
 
-type Parent = ActorRefFrom<TSignInRouterMachine>;
+export type DetermineStartingFactorInput = {
+  parent: SignInRouterMachineActorRef;
+};
 
 export type PrepareFirstFactorInput = WithParams<SignInFirstFactor | null> & {
-  parent: Parent;
+  parent: SignInRouterMachineActorRef;
   resendable: boolean;
 };
 export type PrepareSecondFactorInput = WithParams<SignInSecondFactor | null> & {
-  parent: Parent;
+  parent: SignInRouterMachineActorRef;
   resendable: boolean;
 };
 
-export type AttemptFirstFactorInput = { parent: Parent; fields: FormFields; currentFactor: SignInFirstFactor | null };
-export type AttemptSecondFactorInput = { parent: Parent; fields: FormFields; currentFactor: SignInSecondFactor | null };
+export type AttemptFirstFactorInput = {
+  parent: SignInRouterMachineActorRef;
+  fields: FormFields;
+  currentFactor: SignInFirstFactor | null;
+};
+export type AttemptSecondFactorInput = {
+  parent: SignInRouterMachineActorRef;
+  fields: FormFields;
+  currentFactor: SignInSecondFactor | null;
+};
 
 export const SignInVerificationMachineId = 'SignInVerification';
 
 const SignInVerificationMachine = setup({
   actors: {
+    determineStartingFactor: fromPromise<SignInFactor | null, DetermineStartingFactorInput>(() =>
```

Comment:
The issue I was running into was that, when refreshing the page and when `determineStartingFactor` was an action, because actions are fire-and-forget, it was running before the context was present.

Moving it to an actor ensures that everything we need is present...

**Flow Before:**

```
xstate.init SignInFirstFactorMachine
  entry: determineStartingFactor (as Action)
  initial: 'Preparing'
  step: 'Preparing'
    ...
```

**Flow After:**

```
xstate.init SignInFirstFactorMachine
  initial: 'Init'
  step: Init
    invoke: determineStartingFactor (as Actor)
      onDone: 'Preparing'
  step: 'Preparing'
    ...
```